### PR TITLE
controller: DR volume should wait for the latest backup restored before activation

### PR DESCRIPTION
**Pull request by shuo-wu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/7945

### CODE REVIEW

**Comment by shuo-wu** [comment]
The v1.5.x regression test: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/6302/

File: controller/volume_controller.go [hunk 1]
```diff
@@ -3146,49 +3146,72 @@
...
~		!v.Spec.Standby) {
 		return nil
 	}
 
+	// Make sure the backup volume is up-to-date and the latest backup is restored
+	_, bvName, _, err := backupstore.DecodeBackupURL(v.Spec.FromBackup)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get backup name from volume %s backup URL %v", v.Name, v.Spec.FromBackup)
+	}
+	bv, err := c.ds.GetBackupVolume(bvName)
```
Comment (innobead) [inline controller/volume_controller.go:3181]: `GetBackupVolumeRO`

File: controller/volume_controller.go [hunk 2]
```diff
@@ -3170,6 +3173,26 @@ func (c *VolumeController) checkAndFinishVolumeRestore(v *longhorn.Volume, e *lo
...
+	if err != nil {
+		return err
+	}
+	if !bv.Status.LastSyncedAt.IsZero() &&
+		bv.Spec.SyncRequestedAt.After(bv.Status.LastSyncedAt.Time) {
+		log.Infof("Restore/DR volume needs to wait for backup volume %s update", bvName)
+		return nil
```
Comment (innobead) [inline controller/volume_controller.go:3188]: Should we requeue the volume?

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. Just a few comments.

File: controller/volume_controller.go (hunk 2 above)
Comment (shuo-wu) [reply controller/volume_controller.go:3188]: Here the volume controller should wait for the backup volume controller to sync up this `bv`.

**Comment by shuo-wu** [comment]
@mergify backport v1.6.x v1.5.x

**Pull request merged**